### PR TITLE
Linting and formatting

### LINKED ISSUE #48: Add linting

### PR DESCRIPTION
**Pull request by farski**
Adds general support for linting, formatting, etc across the project, as well as brings existing code into line with the new style guidelines. This is done mainly via Prettier, ESLint, and cfn-python-lint. Those are now all run against the project during CI. The rules/settings/etc for each have been added to the project,

Closes #48

### CODE REVIEW

File: buildspec.yml [hunk 1]
```diff
@@ -1,14 +1,24 @@
...
+      - 'echo "Installing CloudFormation Linter"'
+      - 'pip install cfn-lint --upgrade'
+      - 'echo "Installing Prettier"'
+      - 'npm install'
   build:
     commands:
-      - "cd $(ls -d */|head -n 1)"
+      - 'cfn-lint --ignore-checks W --template template.yml'
+      - 'npm run prettier -- --check "**/*.{js,json,yaml,yml}"'
+      - 'npm run eslint -- "**/*.js"'
```
Comment (cavis) [inline buildspec.yml:21]: Ah, so linting is now part of the build.  But no tests?

File: lambdas/CallbackLambdaFunction/index.js [hunk 2]
```diff
@@ -48,13 +53,15 @@ function httpRequest(event, message, redirectCount) {
...
-              msg: `Following HTTP redirect`,
-              location: res.headers.location,
-              count: redirectCount
-            }));
+            console.log(
+              JSON.stringify({
+                msg: `Following HTTP redirect`,
+                location: res.headers.location,
+                count: redirectCount,
+              }),
```
Comment (cavis) [inline lambdas/CallbackLambdaFunction/index.js:61]: Hmmph... trailing commas, you say.  Not my favorite setting, but a good linter makes everyone unhappy.

File: lambdas/ImageTransformLambdaFunction/index.js [hunk 3]
```diff
@@ -86,16 +93,23 @@ async function s3Upload(event, imageFileTmpPath) {
   // When the optional `ContentType` property is set to `REPLACE`, if a MIME is
   // included with the artifact, that should be used as the new images's
   // content type
-  if (event.Task.Destination.hasOwnProperty('ContentType')
-      && event.Task.Destination.ContentType === 'REPLACE'
-      && event.Artifact.hasOwnProperty('Descriptor')
-      && event.Artifact.Descriptor.hasOwnProperty('MIME')) {
+  if (
+    Object.prototype.hasOwnProperty.call(
```
Comment (cavis) [inline lambdas/ImageTransformLambdaFunction/index.js:97]: Ugh.  Yeah, this syntax.  Not my favorite, but what can you do.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    "eslint": "^6.8.0",
+    "eslint-config-airbnb-base": "^14.1.0",
+    "eslint-config-prettier": "^6.11.0",
+    "eslint-plugin-import": "^2.20.2",
+    "eslint-plugin-prettier": "^3.1.3",
+    "prettier": "^2.0.5"
+  },
+  "scripts": {
+    "eslint": "eslint",
+    "prettier": "prettier"
```
Comment (cavis) [inline package.json:20]: No test script?

Oh, I see, the test directory has its own package.json.  Is that redundant now?  Seems like that should all move up to here.

File: .prettierrc.yml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+semi: true
+trailingComma: 'all'
+singleQuote: true
+printWidth: 80
```
Comment (cavis) [inline .prettierrc.yml:4]: I don't love this.  You've got yml/comments/json all going more like 100 characters, and wrapping some of these js lines at 80 makes them look funky.

File: .prettierrc.yml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+semi: true
+trailingComma: 'all'
```
Comment (cavis) [inline .prettierrc.yml:2]: Also not a huge fan of this.  Just seems odd to throw commas after function call arguments.  But I could probably live with this if I must!

File: .prettierrc.yml (hunk 5 above)
Comment (farski) [reply .prettierrc.yml:4]: Prettier is even more opinionated than usual about [that](https://prettier.io/docs/en/options.html#print-width), so I went with the default.

File: .prettierrc.yml (hunk 6 above)
Comment (farski) [reply .prettierrc.yml:2]: Does it help that you don't actually have to type the commas yourself?

File: package.json (hunk 4 above)
Comment (farski) [reply package.json:20]: I don't look in the test directory anymore.

File: .prettierrc.yml (hunk 6 above)
Comment (cavis) [reply .prettierrc.yml:2]: Yeah, it's fine.  Just fighting years of training my brain to track down trailing commas in javascript errors in IE.

File: .prettierrc.yml (hunk 5 above)
Comment (cavis) [reply .prettierrc.yml:4]: Ugh... yeah, someone on that project really wants to die on that hill.  I'm going to make my own linter that argues for 40 character lines.

**Pull request merged**